src/Symfony/Component/Yaml/Inline.php

```diff
@@ -397,7 +397,7 @@ private static function parseSequence($sequence, $flags, &$i = 0, $references =
                     $value = self::parseScalar($sequence, $flags, array(',', ']'), array('"', "'"), $i, true, $references);
 
                     // the value can be an array if a reference has been resolved to an array var
-                    if (!is_array($value) && !$isQuoted && false !== strpos($value, ': ')) {
+                    if (!is_array($value) && !$value instanceof \DateTimeInterface && !$isQuoted && false !== strpos($value, ': ')) {
```

Comment:
I would replace the first parts by ``is_string($value)``. The strpos condition can pass only for strings anyway, not for integers or floats, so it is useless to try it for them